### PR TITLE
Cleanup schools bug fix

### PR DESCRIPTION
**Pull request by ArushC**
### [Pivotal Tracker Link][tracker]

[tracker]: https://www.pivotaltracker.com/story/show/your-story-id

## What this PR does:

This PR does three main things:

1) fixes the Javascript that toggles the state "required" field when the country selected is/is not the USA, and fixes backend validations on the Schools#new and Schools#create pages.

2) Adds two more backend tests for the international schools feature through the RSpec testing framework.

3) Refactors ALL flash error messages in the application to adopt a standard format: "An error occured: [LIST OF ERRORS THAT OCCURED IN HUMAN-READABLE FORMAT]". I updated the Cucumber/RSpec tests accordingly. The error outputs are now considerably more informative than they were earlier ("Failed to submit information :(" does not tell me *why* the error occured). 

I merged this PR directly into the international schools branch, for which a pull request to the Golden Repo is still open at [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/295](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/295).

This pull request fixes|implements (pick one...) ______.

#### Who authored this PR?

@ArushC 

### How should this PR be tested?

View the latest Heroku deployment, and look at the RSpec/Cucumber tests.

* Is there a deploy we can view?

Yes:  [ https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/](https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/)

EDIT: the current Heroku deployment contains the languages feature, not this one. Review that PR before this one and then force push this branch to the main on Heroku to deploy this.

* What do the specs/features test?
* Are there edge cases to watch out for?

#### Are there any complications to deploying this?

None

### Checklist:

- [x] Has this been deployed to a staging environment or reviewed by a customer?
- [x] Tag someone for code review (either a coach / team member)
- [x] I have renamed the branch to match PivotTracker's suggested one (necessary for BlueJay) (e.g. `michael/12345-add-new-feature` Any branch name will do as long as the story ID is there. You can use `git checkout -b [new-branch-name]`)

### CODE REVIEW

**Comment by ArushC** [comment]
@perryzjc @razztech @JacksonXu33 @realmichaeltao someone please review this PR

**Comment by perryzjc** [comment]
@ArushC Nice Job. I finished the review for this PR and pushed some commits to fix some of the issues. I also added a new test. There was one tricky thing my personal computer hard to verify. Take a look at my review for `render 'new' && return` above.

**Comment by ArushC** [comment]
@perryzjc "Take a look at my review for render 'new' && return above" --> what does this mean? I don't see any review.

**Comment by perryzjc** [comment]
> @perryzjc "Take a look at my review for render 'new' && return above" --> what does this mean? I don't see any review.

@ArushC Is this screenshot visible on your side?
image

**Comment by ArushC** [comment]
wait that's so weird I literally don't see any review here... hmmm... ok well I'll take a look at that

File: app/controllers/schools_controller.rb [hunk 1]
```diff
@@ -45,10 +45,11 @@ def update
     @school = School.find(params[:id])
     @school.assign_attributes(school_params)
     if @school.save
-      flash[:success] = "Update #{@school.name} successfully."
+      flash[:success] = "Updated #{@school.name} successfully."
```
Comment (fp456) [inline app/controllers/schools_controller.rb:48]: From codecov: Added line #L48 was not covered by tests

File: app/controllers/schools_controller.rb [hunk 2]
```diff
@@ -45,10 +45,11 @@ def update
...
       redirect_to school_path(@school)
     else
-      render "edit", alert: "Failed to submit information :("
+      flash[:alert] = "An error occurred: #{@school.errors.full_messages.join(', ')}"
```
Comment (fp456) [inline app/controllers/schools_controller.rb:51]: Added lines #L51 - L52 were not covered by tests

File: app/controllers/teachers_controller.rb [hunk 3]
```diff
@@ -81,7 +81,7 @@ def create
       TeacherMailer.teacher_form_submission(@teacher).deliver_now
       redirect_to root_path
     else
-      redirect_to new_teacher_path, alert: "An error occurred while trying to save. #{@teacher.errors.full_messages}"
+      redirect_to new_teacher_path, alert: "An error occurred: #{@teacher.errors.full_messages.join(', ')}"
```
Comment (fp456) [inline app/controllers/teachers_controller.rb:84]: Added line #L84 was not covered by tests

**Review by fp456 — COMMENTED** [review]
LGTM but for the codecov alerts, I'm curious why there's no tests for the lines highlighted. Is it because it can't be tested? If it's feasible it would be reasonable to test that too. You can do it in another PR later if that's more convenient. Good job
